manager/container.go

```diff
@@ -181,20 +181,27 @@ func (c *containerData) housekeeping() {
 
 		// Log usage if asked to do so.
 		if c.logUsage {
-			stats, err := c.storageDriver.RecentStats(c.info.Name, 2)
+			const numSamples = 60
+			stats, err := c.storageDriver.RecentStats(c.info.Name, numSamples)
 			if err != nil {
 				if c.allowErrorLogging() {
 					glog.Infof("[%s] Failed to get recent stats for logging usage: %v", c.info.Name, err)
 				}
-			} else if len(stats) < 2 {
+			} else if len(stats) < numSamples {
 				// Ignore, not enough stats yet.
 			} else {
-				usageCpuNs := stats[1].Cpu.Usage.Total - stats[0].Cpu.Usage.Total
-				usageMemory := stats[1].Memory.Usage
+				usageCpuNs := uint64(0)
+				for i := range stats {
+					if i > 0 {
+						usageCpuNs += (stats[i].Cpu.Usage.Total - stats[i-1].Cpu.Usage.Total)
+					}
+				}
+				usageMemory := stats[numSamples-1].Memory.Usage
 
-				usageInCores := float64(usageCpuNs) / float64(stats[1].Timestamp.Sub(stats[0].Timestamp).Nanoseconds())
+				instantUsageInCores := float64(stats[numSamples-1].Cpu.Usage.Total-stats[numSamples-2].Cpu.Usage.Total) / float64(stats[numSamples-1].Timestamp.Sub(stats[numSamples-2].Timestamp).Nanoseconds())
```

Comment:
Instead, how about printing max and maybe 90%ile which might be more
useful. The reason I say is that this instantaneous value only looks at 1
second period every 60 seconds and does not care about cadvisor's cpu usage
during the other 59 seconds in the same period.

On Mon, Feb 9, 2015 at 1:46 PM, Victor Marmol <notifications@github.com>
wrote:

> In manager/container.go
> <https://github.com/google/cadvisor/pull/497#discussion_r24368424>:
>
> >
> > -				usageInCores := float64(usageCpuNs) / float64(stats[1].Timestamp.Sub(stats[0].Timestamp).Nanoseconds())
> > +				instantUsageInCores := float64(stats[numSamples-1].Cpu.Usage.Total-stats[numSamples-2].Cpu.Usage.Total) / float64(stats[numSamples-1].Timestamp.Sub(stats[numSamples-2].Timestamp).Nanoseconds())
>
> This prints the instantaneous usage and then we show the average over the
> last 60s.
>
> It is useful since it gives us an idea of the current usage (and how
> stable that is) as well as what it means in the "medium" (60s) term overall.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/google/cadvisor/pull/497/files#r24368424>.
>